hybridse/include/node/sql_node.h

```diff
@@ -1115,48 +1116,45 @@ class FrameBound : public SqlNode {
 
     ~FrameBound() {}
 
-    void Print(std::ostream &output, const std::string &org_tab) const {
-        SqlNode::Print(output, org_tab);
-        const std::string tab = org_tab + INDENT + SPACE_ED;
-        std::string space = org_tab + INDENT + INDENT;
-        output << "\n";
-        output << tab << SPACE_ST << "bound: " << BoundTypeName(bound_type_) << "\n";
+    void Print(std::ostream &output, const std::string &org_tab) const override;
 
-        if (kFollowing == bound_type_ || kPreceding == bound_type_) {
-            output << space << offset_;
-        }
-    }
     const std::string GetExprString() const {
-        switch (bound_type_) {
-            case node::kCurrent:
-                return "0";
-            case node::kOpenFollowing:
-            case node::kOpenPreceding:
-            case node::kFollowing:
-            case node::kPreceding:
-                return std::to_string(GetSignedOffset());
-            case node::kPrecedingUnbound:
-            case node::kFollowingUnbound:
-                return "UNBOUND";
-        }
-        return "";
+        return absl::StrCat(offset_, " ", BoundTypeName(bound_type_));
     }
 
     BoundType bound_type() const { return bound_type_; }
     const bool is_time_offset() const { return is_time_offset_; }
     int64_t GetOffset() const { return offset_; }
-    int64_t GetSignedOffset() const {
+
+
+    /// \brief get the inclusive frame bound offset value that has signed symbol
+    ///
+    /// usually, `offset_` is non-negative (think about the offset expr in SQL window frame:
+    ///         'ROWS BETWEEN 2 PRECEDING AND 0 OPEN PRECEDING')
+    /// * for 'PRECEDING' bound, output is negative
+    /// * for 'FOLLOWING' bound, output is positive
+    /// * for 'OPEN [ PRECEDING | FOLLOWING ]', output is one increasing or decreasing, based on
+    ///   whether it is start bound or end bound
+    ///
+    /// \param is_start_frame whether this bound is start or end of window frame
+    ///
+    /// by convention, the frame that is more close to the current row is the end frame
```

Comment:
is it the earliest frame is start frame? e.g.,
`2 FOLLOWING and 5 FOLLOWING`
`2 FOLLOWING` is the start frame and `5 FOLLOWING` is the end frame?

Reply:
Should be, the lower offset is the start frame. And strictly, the `SignedOffset` of start frame should less or equal to end frame. see #1951